test-framework/remote/src/main/java/org/keycloak/testframework/runonserver/RunOnServerSupplier.java

```diff
@@ -0,0 +1,63 @@
+package org.keycloak.testframework.runonserver;
+
+import org.apache.http.client.HttpClient;
+import org.keycloak.testframework.injection.InstanceContext;
+import org.keycloak.testframework.injection.LifeCycle;
+import org.keycloak.testframework.injection.RequestedInstance;
+import org.keycloak.testframework.injection.Supplier;
+import org.keycloak.testframework.injection.SupplierOrder;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.remote.RemoteProviders;
+import org.keycloak.testframework.server.KeycloakUrls;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RunOnServerSupplier implements Supplier<RunOnServerClient, InjectRunOnServer> {
+
+    @Override
+    public Class<InjectRunOnServer> getAnnotationClass() {
+        return InjectRunOnServer.class;
+    }
+
+    @Override
+    public Class<RunOnServerClient> getValueType() {
+        return RunOnServerClient.class;
+    }
+
+    @Override
+    public Set<Class<?>> dependencies() {
+        return Set.of(HttpClient.class, ManagedRealm.class, RemoteProviders.class, KeycloakUrls.class);
+    }
+
+    @Override
+    public RunOnServerClient getValue(InstanceContext<RunOnServerClient, InjectRunOnServer> instanceContext) {
+        HttpClient httpClient = instanceContext.getDependency(HttpClient.class);
+        ManagedRealm realm = instanceContext.getDependency(ManagedRealm.class, instanceContext.getAnnotation().realmRef());
+        KeycloakUrls keycloakUrls = instanceContext.getDependency(KeycloakUrls.class);
+        instanceContext.getDependency(RemoteProviders.class);
+
+        TestClassServer testClassServer = instanceContext.getDependency(TestClassServer.class);
+        String[] permittedPackages = instanceContext.getAnnotation().permittedPackages();
+        testClassServer.addPermittedPackages(new HashSet<>(Arrays.asList(permittedPackages)));
+
+        return new RunOnServerClient(httpClient, keycloakUrls.getBase(), realm.getName());
```

Comment:
Instead of getting the base URL from `KeycloakUrls` and then gluing it together with the realm name later, we could get it at once here from `realm.getBaseUrl()`

Reply:
Good point, one less dependency as well. Updated